sw/host/opentitanlib/src/image/image.rs

```diff
@@ -102,120 +104,110 @@ impl Image {
         Ok(())
     }
 
-    /// Updates the data in a pre-existing extension. Does not update the image size.
-    pub fn update_extension_data(&mut self, id: u32, bytes: &[u8]) -> Result<()> {
-        let entry = self
-            .borrow_manifest()?
-            .extensions
-            .entries
-            .iter()
-            .find(|x| x.identifier == id)
-            .ok_or(ImageError::NoExtensionTableEntry(id))?;
-        let offset = entry.offset as usize + std::mem::size_of::<ManifestExtHeader>();
-        let extension_slice = self
-            .data
-            .bytes
-            .get_mut(offset..offset + bytes.len())
-            .ok_or(ImageError::ExtensionOverflow)?;
-        extension_slice.copy_from_slice(bytes);
+    /// Adds an extension to the signed region of this `Image`.
```

Comment:
nit: can you add a brief comment that explains why we need this? (SPX+ signature is an unsigned extension but requires the signed region to be finalized).